test_particlef and cgwrite_f03 tests fail on many architectures

We are trying to build packages of CGNS 4.5.0 for debian linux.
Build results can be found at https://buildd.debian.org/status/package.php?p=libcgns&suite=experimental
amd64 and arm64 and several other architectures build and pass tests fine.

But many other architectures fail `test_particlef` and/or `cgwrite_f03`. More often `test_particlef`.
The debian libcgns package uses the serial build of libhdf5 (1.14.5), so the problem is not driven by MPI. 
The compiler is gcc 15.2.0.

[armhf](https://buildd.debian.org/status/fetch.php?pkg=libcgns&arch=armhf&ver=4.5.0-1%7Eexp1&stamp=1760289552&raw=0), for instance, reports the `test_particlef` error:
```
test 34
      Start 34: test_particlef

34: Test command: /build/reproducible-path/libcgns-4.5.0/obj-arm-linux-gnueabihf/src/tests/test_particlef
34: Working Directory: /build/reproducible-path/libcgns-4.5.0/obj-arm-linux-gnueabihf/src/tests
34: Environment variable modifications: 
34:  PATH=path_list_prepend:
34: Test timeout computed to be: 10000000
34: creating CGNS file test_particlef.cgns
34:    1.00000005E-03   1.00000005E-03
34: creating CGNS file particle_partial_testf.cgns
34:  writing particle zone
34:  coordinates -> 1,3,5,7 ...
34: field -> 1,3,5,7 ...
34:  closing and reopening in modify mode
34:  coordinates -> 2,4,6,8 ...
34:  field -> 2,4,6,8 ...
34:  closing and reopening in read mode
34: creating CGNS file particle_bboxf.cgns
34:  writing particle zone
34:  closing and reopening in read mode
34:  Error: No bounding box found for reading
34:  closing and reopening in modify mode
34:  closing and reopening in read mode
34:  closing and reopening to test limit cases
34:  Error: Invalid data type for bounding box array: 2
34:  closing file
34:  creating CGNS file particle_coord_and_ptset_testf.cgns
34:  writing particle zone
34:  writing coordinate nodes
34:  closing and reopening in read mode
34: *** stack smashing detected ***: terminated
34: 
34: Program received signal SIGABRT: Process abort signal.
34: 
34: Backtrace for this error:
34/34 Test #34: test_particlef ...................Subprocess aborted***Exception:   0.12 sec
creating CGNS file test_particlef.cgns
   1.00000005E-03   1.00000005E-03
creating CGNS file particle_partial_testf.cgns
 writing particle zone
 coordinates -> 1,3,5,7 ...
field -> 1,3,5,7 ...
 closing and reopening in modify mode
 coordinates -> 2,4,6,8 ...
 field -> 2,4,6,8 ...
 closing and reopening in read mode
creating CGNS file particle_bboxf.cgns
 writing particle zone
 closing and reopening in read mode
 Error: No bounding box found for reading
 closing and reopening in modify mode
 closing and reopening in read mode
 closing and reopening to test limit cases
 Error: Invalid data type for bounding box array: 2
 closing file
 creating CGNS file particle_coord_and_ptset_testf.cgns
 writing particle zone
 writing coordinate nodes
 closing and reopening in read mode
*** stack smashing detected ***: terminated

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:


97% tests passed, 1 tests failed out of 34

Total Test time (real) =   6.51 sec

The following tests FAILED:
	 34 - test_particlef (Subprocess aborted)
Errors while running CTest
make[2]: *** [Makefile:94: test] Error 8
```

The [s390x](https://buildd.debian.org/status/fetch.php?pkg=libcgns&arch=s390x&ver=4.5.0-1%7Eexp1&stamp=1760290130&raw=0) build reports a `SIGSEGV: Segmentation fault - invalid memory reference` for `test_particlef`, but for `cgwrite_f03` not so much much information is reported,
```
test 26
      Start 26: cgwrite_f03

26: Test command: /build/reproducible-path/libcgns-4.5.0/obj-s390x-linux-gnu/src/tests/cgwrite_f03
26: Working Directory: /build/reproducible-path/libcgns-4.5.0/obj-s390x-linux-gnu/src/tests
26: Environment variable modifications: 
26:  PATH=path_list_prepend:
26: Test timeout computed to be: 10000000
26: no CGNS error reported
26/34 Test #26: cgwrite_f03 ......................***Failed    0.01 sec
no CGNS error reported
``` 
[sparc64](https://buildd.debian.org/status/fetch.php?pkg=libcgns&arch=sparc64&ver=4.5.0-1%7Eexp1&stamp=1760292507&raw=0) reports the `cgwrite_f03` error with `SIGBUS: Access to an undefined portion of a memory object`

--- a/src/cgns_f.F90
+++ b/src/cgns_f.F90
@@ -6701,11 +6701,11 @@ CONTAINS
     INTEGER, INTENT(IN) :: fn
     INTEGER, INTENT(IN) :: B
     INTEGER, INTENT(IN) :: P
-    CHARACTER(LEN=*) , INTENT(INOUT) :: particlename
+    CHARACTER(LEN=*) , INTENT(OUT) :: particlename
     INTEGER(CGSIZE_T), INTENT(OUT)   :: nsize
     INTEGER, INTENT(OUT) :: ier
 
-    CHARACTER(LEN=LEN_TRIM(particlename)+1,KIND=C_CHAR) :: c_particlename
+    CHARACTER(len=1, kind=C_CHAR) :: c_particlename(MAX_LEN+1)
 
     INTERFACE
        INTEGER(C_INT) FUNCTION cg_particle_read(fn, B, P, particlename, nsize) BIND(C, NAME="cg_particle_read")
@@ -6826,10 +6826,10 @@ CONTAINS
       INTEGER, INTENT(IN) :: B
       INTEGER, INTENT(IN) :: P
       INTEGER, INTENT(IN) :: C
-      CHARACTER(LEN=*) :: pcoord_name
+      CHARACTER(LEN=*), INTENT(OUT) :: pcoord_name
       INTEGER, INTENT(OUT) :: ier
 
-      CHARACTER(LEN=LEN_TRIM(pcoord_name)+1, KIND=C_CHAR) :: c_pcoord_name
+      CHARACTER(len=1, kind=C_CHAR) :: c_pcoord_name(MAX_LEN+1)
 
       INTERFACE
          INTEGER(C_INT) FUNCTION cg_particle_coord_node_read(fn, B, P, C, pcoord_name) BIND(C, NAME="cg_particle_coord_node_read")
@@ -6987,11 +6987,11 @@ CONTAINS
       INTEGER, INTENT(IN) :: P
       INTEGER, INTENT(IN) :: C
       INTEGER(cgenum_t), INTENT(OUT) :: datatype
-      CHARACTER(LEN=*)     :: coordname
+      CHARACTER(LEN=*), INTENT(OUT) :: coordname
       INTEGER, INTENT(OUT) :: ier
 
       INTEGER(C_INT) :: c_datatype
-      CHARACTER(LEN=LEN_TRIM(coordname)+1, KIND=C_CHAR) :: c_coordname
+      CHARACTER(len=1, kind=C_CHAR) :: c_coordname(MAX_LEN+1)
 
       INTERFACE
           INTEGER(C_INT) FUNCTION cg_particle_coord_info(fn, B, P, C, datatype, coordname) BIND(C, NAME="cg_particle_coord_info")
@@ -7219,10 +7219,10 @@ CONTAINS
       INTEGER, INTENT(IN) :: B
       INTEGER, INTENT(IN) :: P
       INTEGER, INTENT(IN) :: S
-      CHARACTER(LEN=*)    :: solname
+      CHARACTER(LEN=*), INTENT(OUT) :: solname
       INTEGER, INTENT(OUT) :: ier
 
-      CHARACTER(LEN=LEN_TRIM(solname)+1,KIND=C_CHAR) :: c_solname
+      CHARACTER(len=1, kind=C_CHAR) :: c_solname(MAX_LEN+1)
 
       INTERFACE
           INTEGER(C_INT) FUNCTION cg_particle_sol_info(fn, B, P, S, solname) BIND(C, NAME="cg_particle_sol_info")
@@ -7481,10 +7481,10 @@ CONTAINS
       INTEGER, INTENT(IN)  :: S
       INTEGER, INTENT(IN)  :: F
       INTEGER(cgenum_t), INTENT(OUT) :: datatype
-      CHARACTER(LEN=*),  INTENT(INOUT) :: fieldname
+      CHARACTER(LEN=*),  INTENT(OUT) :: fieldname
       INTEGER, INTENT(OUT) :: ier
 
-      CHARACTER(LEN=LEN_TRIM(fieldname)+1, KIND=C_CHAR) :: c_fieldname
+      CHARACTER(len=1, kind=C_CHAR) :: c_fieldname(MAX_LEN+1)
 
      INTERFACE
           INTEGER(C_INT) FUNCTION cg_particle_field_info(fn, B, P, S, F, datatype, fieldname) BIND(C, NAME="cg_particle_field_info")
@@ -7635,10 +7635,10 @@ CONTAINS
       INTEGER, INTENT(IN) :: fn
       INTEGER, INTENT(IN) :: B
       INTEGER, INTENT(IN) :: P
-      CHARACTER(LEN=*),  INTENT(INOUT) :: pitername
+      CHARACTER(LEN=*), INTENT(OUT) :: pitername
       INTEGER, INTENT(OUT) :: ier
 
-      CHARACTER(LEN=LEN_TRIM(pitername)+1, KIND=C_CHAR) :: c_pitername
+      CHARACTER(len=1, kind=C_CHAR) :: c_pitername(MAX_LEN+1)
 
       INTERFACE
          INTEGER(C_INT) FUNCTION cg_piter_read(fn, B, P, pitername) &

--- a/src/cgns_f.F90
+++ b/src/cgns_f.F90
@@ -7824,11 +7824,11 @@ CONTAINS
 !DEC$endif
     SUBROUTINE cg_particle_model_read_f(ModelLabel, ModelType, ier)
       IMPLICIT NONE
-      CHARACTER(LEN=*) , INTENT(INOUT) :: ModelLabel
+      CHARACTER(LEN=*), INTENT(OUT) :: ModelLabel
       INTEGER(CGENUM_T), INTENT(OUT) :: ModelType
       INTEGER, INTENT(OUT) :: ier
 
-      CHARACTER(LEN=LEN_TRIM(ModelLabel)+1,KIND=C_CHAR) :: c_ModelLabel
+      CHARACTER(len=1, kind=C_CHAR) :: c_ModelLabel(MAX_LEN+1)
 
       INTERFACE
          INTEGER(C_INT) FUNCTION cg_particle_model_read(ModelLabel, ModelType) &

